src/detail/header.cpp

```diff
@@ -597,11 +597,30 @@ on_insert_transfer_encoding()
         md.transfer_encoding.codings += rv->size();
         for(auto t : *rv)
         {
-            if(! md.transfer_encoding.is_chunked)
+            auto& mte = md.transfer_encoding;
```

Comment:
What about Content-Encoding?

Reply:
We should probably just add a corresponding `on_insert_content_encoding` that gets invoked.
We need to also reconcile what happens when a user does:
```
content-encoding: deflate\r\n
transfer-encoding: deflate\r\n
```
Maybe we just simply treat this as an error.